### PR TITLE
add: example on semantic segmentation.

### LINKED ISSUE #45: Add an example demoing the use of PEFT for SegFormer Semantic Segmentation
Similar to #44 but for SegFormer (semantic segmentation). 

@pacman100

### PR DESCRIPTION
**Pull request by sayakpaul**
Closes #45 

There are two minor changes to the image classification example. I know this is not a good PR practice but the changes were extremely small.

### CODE REVIEW

**Comment by pacman100** [comment]
wrt semantic segmentation example:
1. can we remove the cell ouputs of trainer.train() as they are huge and will take up considerable space
2. I think the example notebook has bug because `Accuracy Wall` is `nan`  for full finetuning too but in [stevhliu/my_awesome_seg_model · Hugging Face](https://huggingface.co/stevhliu/my_awesome_seg_model) it's very high. I tried full fine-tuning using 50 images as the task page (40 train + 10 eval)

**Comment by sayakpaul** [comment]
I will address the first point in the follow-up commits. Regarding the second point, I think there are also quite a few nan values for the other classes in the repo you mentioned. 

If you take a look at the training loss, it looks similar, though. I think the category that might be impacted because of the low number of training examples depends on their quantity in the training set as well. Not sure what you mean by the bug.

**Comment by pacman100** [comment]
By bug I mean I'm unable to reproduce the same results for full fine-tuning using same train samples and eval samples along with same trainer arguments as that task page repo

**Comment by pacman100** [comment]
this is what I see for full fine-tuning: https://github.com/pacman100/temp/blob/main/Copy_of_semantic_segmentation_peft_lora%20(1).ipynb

Screenshot 2023-02-08 at 12 14 40 PM

**Comment by sayakpaul** [comment]
@pacman100 addressed the things we discussed during the call. PTAL.

**Comment by pacman100** [comment]
Thank you for iterating! LGTM 🤗
